compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -4598,19 +4598,41 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
             return;
         }
 
+        fn get_fn_decl<'tcx, 'hir>(
```

Comment:
Could you rename this method? The first reaction I had seeing it was to suggest using `node.fn_decl()` and `node.body_id()`, but that's not what this function does: it chooses on which kind of items emit the diagnostic.

+ Could you add a few comments explaining why only these cases? Why not foreign fns? Why not trait decls / trait impls.

Reply:
> Could you rename this method? The first reaction I had seeing it was to suggest using `node.fn_decl()` and `node.body_id()`, but that's not what this function does: it chooses on which kind of items emit the diagnostic.
> 
> * Could you add a few comments explaining why only these cases? Why not foreign fns? Why not trait decls / trait impls.

Thank you.  Sorry for the slow response.
Hello, this is introduced to solve issue # 125997, mainly suggestion for beginners.
It recommend adding a return type to the function definition. To prevent new errors, a corresponding return value needs to be added at the end of the function body.
For beginners, I hope to fix it directly here to avoid introducing new errors and requiring them to add return values by themselves. 
I am concerned that the trait mehod may be used elsewhere, and modifying the function signature may result in errors in other trait impls. I think for foreign fn, we couldn't add a return value at the end of the function body.

Reply:
Please put summarized comments in the code. Having the explanation on github won't help the next person reading the code.

Reply:
> Please put summarized comments in the code. Having the explanation on github won't help the next person reading the code.

Done. Thank you.